packages/rn-tester/js/examples/Modal/ModalPresentation.js

```diff
@@ -59,7 +87,34 @@ function ModalPresentation() {
     if (action === 'onShow') {
       alert('onShow');
     }
+    if (ref != null) {
+      setModalOpened(true);
+    }
   };
+
+  React.useEffect(() => {
+    let timer;
+    if (ref != null && modalOpened === true) {
+      if (Platform.OS === 'ios') {
+        // $FlowFixMe
+        AccessibilityInfo.sendAccessibilityEvent(ref.current, 'focus');
+      } else {
+        // see https://github.com/facebook/react-native/issues/30097#issuecomment-1285927266
+        timer = setTimeout(() => {
+          if (ref.current != null) {
+            AccessibilityInfo.sendAccessibilityEvent(ref.current, 'focus');
+          }
+        }, 1000);
```

Comment:
It's usually a red flag when I see `setTimeout` has to be used to work around async mounting issue. Is this something we can `useLayoutEffect`?

Reply:
My solution would be to modify the native implementation of sendAccessibilityEvent, to fix the original [Talkback issue](https://stackoverflow.com/q/28472985/7295772). The issue is https://github.com/facebook/react-native/issues/30097, but it is not in my sprint. 

The Pull Request was meant to be an API proposal.
I published the API proposal to receive feedback on my comment https://github.com/facebook/react-native/pull/34969#issuecomment-1320258021.

I started testing a solution to the native method sendAccessibilityEvent in commit https://github.com/facebook/react-native/pull/34969/commits/17d78273dbfffd68c66cd5bab82f4a68e236c3dc#diff-d3d0a4709d26919cf2a549febf6cdb755a1274a489078154a80c04d4ba82e590R969-R980, I can publish and improve the solution in a separate PR (for ex. an alternative implementation of [sendAccessibilityEvent](https://github.com/facebook/react-native/blob/17d78273dbfffd68c66cd5bab82f4a68e236c3dc/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java#L952-L981) in SurfaceMountingManager). 

I can also work on improving this example or publish an example in the docs. Thanks.

Reply:
@ryancat My solution would be to modify the native implementation of sendAccessibilityEvent, to fix the original [Talkback issue](https://stackoverflow.com/q/28472985/7295772). The issue is https://github.com/facebook/react-native/issues/30097, but it is not in my sprint. 

The Pull Request was meant to be an API proposal.
I published the API proposal to receive feedback on my comment https://github.com/facebook/react-native/pull/34969#issuecomment-1320258021.

I started testing a solution to the native method sendAccessibilityEvent in commit https://github.com/facebook/react-native/pull/34969/commits/17d78273dbfffd68c66cd5bab82f4a68e236c3dc#diff-d3d0a4709d26919cf2a549febf6cdb755a1274a489078154a80c04d4ba82e590R969-R980, I can publish and improve the solution in a separate PR (for ex. an alternative implementation of [sendAccessibilityEvent](https://github.com/facebook/react-native/blob/17d78273dbfffd68c66cd5bab82f4a68e236c3dc/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java#L952-L981) in SurfaceMountingManager). 

I can also work on improving this example or publish an example in the docs. Thanks.

Reply:
While I agree that generally speaking setTimeout is a red flag, in this case we're actually replicating behavior internal to Talkback that works the same way.

When Talkback gets a new "TYPE_WINDOW_STATE_CHANGED" event, it actually waits 550ms before notifying the rest of the system about it, to make sure that the window itself is in a final, idle, state. It also has an up to 1110ms delay while waiting for the screen to be stabilized before responding to that change. So this sort of setTimeout approach is unfortunately not uncommon in the accessibility space.

Since we're in Javascript here, we can't just send the native Android event here, but if there is a solution on the native Android side I think simply firing a "TYPE_WINDOW_STATE_CHANGED" when the modal appears would result in the same behavior of focus moving into it after a slight (talkback controlled) delay. This would leave the ugly setTimeout code to Talkback itself, so that we don't end up in a weird race condition later on if Talkback changes its internal logic.

Some code pointers that may be helpful:
- 550ms delay for responding to window events - https://github.com/google/talkback/blob/acd0bc7631a3dfbcf183789c7557596a45319e1f/talkback/src/main/java/focusmanagement/interpreter/ScreenStateMonitor.java#L209
- 4 200ms retries to wait for the screen to be stablized before responding - https://github.com/google/talkback/blob/acd0bc7631a3dfbcf183789c7557596a45319e1f/talkback/src/main/java/focusmanagement/interpreter/ScreenStateMonitor.java#L226

Reply:
@blavalla Thanks. You are correct. I was able to fix the issue by adding `AccessibilityEvent.TYPE_VIEW_HOVER_ENTER` to the API `sendAccessibilityEvent`.  More info in https://github.com/facebook/react-native/pull/34969#issuecomment-1329779921